src/metakb/harvesters/base.py

```diff
@@ -11,14 +12,23 @@
 class Harvester:
     """A base class for content harvesters."""
 
-    def harvest(self) -> bool:
-        """Retrieve and store records from a resource. Records may be stored in
-        any manner, but must be retrievable by :method:`iterate_records`.
+    def harvest(self, harvested_filepath: str | None = None) -> None:
+        """Retrieve records from a resource and store harvested data in a JSON file.
 
-        :return: `True` if operation was successful, `False` otherwise.
-        :rtype: bool
+        :param harvested_filepath: Path to the JSON file where the harvested data will
+            be stored. If not provided, will use the default path of
+            ``<APP_ROOT>/data/<src_name>/harvester/<src_name>_harvester_YYYYMMDD.json``
+        """
+        harvested_data = self.get_harvester_data()
+        self.create_json(harvested_data, harvested_filepath=harvested_filepath)
+
+    @abstractmethod
+    def get_harvester_data(self) -> dict[str, list[dict]]:
```

Comment:
Naming is hard. This or something else?

Reply:
I think this is a hard thing to name because the `harvest()` method skips a step in terms of logical data processing. You'd expect something like

```
acquire data -> transform it -> return it -> (maybe) save it
```

but instead `harvest` acquires, transforms, and saves it, and then `get_harvester_data` returns the state of the data from right before `harvest` saved it.

anyway I think this is fine the way it is -- maybe rename `harvest()` to `harvest_to_file` so that there's a clearer distinction between what these methods do?